### PR TITLE
Updated the vss-api-netcore to consume the latest controller changes …

### PR DESCRIPTION
**Pull request by chshrikh**
Updated the vss-api-netcore to consume the latest controller changes done for renaming the machine group to deployment group

### CODE REVIEW

File: src/Agent.Listener/Configuration/ConfigurationProvider.cs [hunk 1]
```diff
@@ -263,7 +263,7 @@ public void UpdateAgentSetting(AgentSettings settings)
         {
             ArgUtil.NotNull(_machineGroupServer, nameof(_machineGroupServer));
 
-            DeploymentMachineGroup machineGroup = (await _machineGroupServer.GetDeploymentMachineGroupsAsync(projectName, machineGroupName)).FirstOrDefault();
+            var machineGroup = (await _machineGroupServer.GetDeploymentGroupsAsync(projectName, machineGroupName)).FirstOrDefault();
```
Comment (kmkumaran) [inline src/Agent.Listener/Configuration/ConfigurationProvider.cs]: change the variable to deploymentGroup as well.

File: src/Agent.Listener/Configuration/ConfigurationProvider.cs [hunk 2]
```diff
@@ -263,18 +263,18 @@ public void UpdateAgentSetting(AgentSettings settings)
         {
             ArgUtil.NotNull(_machineGroupServer, nameof(_machineGroupServer));
 
-            DeploymentMachineGroup machineGroup = (await _machineGroupServer.GetDeploymentMachineGroupsAsync(projectName, machineGroupName)).FirstOrDefault();
+            var deploymentGroup = (await _machineGroupServer.GetDeploymentGroupsAsync(projectName, machineGroupName)).FirstOrDefault();
```
Comment (KrishnaAdityaB) [inline src/Agent.Listener/Configuration/ConfigurationProvider.cs]: I am assuming the scope of this PR is to consume the new API only. 
Renaming the user facing inputs and logs will be taken in another PR.

Comment (chshrikh) [inline src/Agent.Listener/Configuration/ConfigurationProvider.cs]: yes  just waiting for RM M113 deployment to finish. Will wait till all UI component get fixed first and then I will work to make changes with agent code

**Comment by TingluoHuang** [comment]
Make sure you run a build/release, and nothing broken because of the new package.

**Comment by chshrikh** [comment]
@TingluoHuang Yes I did the validation with M114. I will validate it with master as well.
